routes/utils.js

```diff
@@ -0,0 +1,85 @@
+function findPrevItemIndex(items, index) {
```

Comment:
Only suggestion is to move this file to `logic/linkedListImplementation.js` or something similar.